### PR TITLE
Fix OpenSSL compilation for tvOS. Fix PNG URL.

### PR DESCRIPTION
**Pull request by pyrosphere**
This PR fixes the compilation of OpenSSL for tvOS.
It also fixes the PNG source download URL from SourceForge.

### CODE REVIEW

File: contrib/src/openssl/rules.mak [hunk 1]
```diff
@@ -107,12 +107,27 @@ export CROSS_SDK=iPhone${IOS_PLATFORM}.sdk
 endif
 
 ifdef HAVE_TVOS
+
 ifeq ($(MY_TARGET_ARCH),arm64)
-OPENSSL_CONFIG_VARS=ios64-cross
```
Comment (minggo) [inline contrib/src/openssl/rules.mak:111]: does it compatible with iOS?

File: contrib/src/openssl/rules.mak [hunk 2]
```diff
@@ -107,12 +107,27 @@ export CROSS_SDK=iPhone${IOS_PLATFORM}.sdk
...
+TVOS_PLATFORM=OS
+OPENSSL_CONFIG_VARS=tvos64-cross-arm64
+OPENSSL_EXTRA_CONFIG_2=no-async
 endif
 ifeq ($(MY_TARGET_ARCH),x86_64)
-OPENSSL_CONFIG_VARS=darwin64-x86_64-cc
```
Comment (minggo) [inline contrib/src/openssl/rules.mak:114]: does it compatible with mac os?

File: contrib/src/openssl/rules.mak (hunk 1 above)
Comment (pyrosphere) [reply contrib/src/openssl/rules.mak:111]: Yes, this line was and still is under `ifdef HAVE_TVOS`.

File: contrib/src/openssl/rules.mak (hunk 2 above)
Comment (pyrosphere) [reply contrib/src/openssl/rules.mak:114]: Yes, this line was and still is under `ifdef HAVE_TVOS`.

File: contrib/src/openssl/rules.mak (hunk 1 above)
Comment (minggo) [reply contrib/src/openssl/rules.mak:111]: but `ios64-cross` is needed to compile for iOS.

Comment (pyrosphere) [reply contrib/src/openssl/rules.mak:111]: The compilation for iOS remains unchanged a little higher on the file. This code only affects tvOS, as it was under `ifdef HAVE_TVOS` but specified the wrong configs (`ios64-cross` and `darwin64-x86_64-cc`). The `ios64-cross` is [still used for iOS](https://github.com/pyrosphere/cocos2d-x-3rd-party-libs-src/blob/9add252849d74be334fbabaaa9c8ce3ebdd07730/contrib/src/openssl/rules.mak#L77).

Comment (minggo) [reply contrib/src/openssl/rules.mak:111]: Oh, sorry, i didn't notice it is under `HAVE_TVOS `. Just notice the modification from `OPENSSL_CONFIG_VARS=ios64-cross` to `TVOS_PLATFORM=OS`.

**Pull request merged**